### PR TITLE
Several caprevoke fixes

### PR DESCRIPTION
**Pull request by markjdb**
No regression tests yet, but I've been writing one which tries to exercise load-cap fault handling on a superpage.

### CODE REVIEW

File: sys/vm/vm_fault.c [hunk 1]
```diff
@@ -955,6 +955,8 @@ vm_fault_trap(vm_map_t map, vm_offset_t vaddr, vm_prot_t fault_type,
 		case KERN_PROTECTION_FAILURE:
 			if ((fault_type & VM_PROT_WRITE_CAP) != 0)
 				segv_ucode = SEGV_STORETAG;
+			else if ((fault_type & VM_PROT_READ_CAP) != 0)
+				segv_ucode = SEGV_LOADTAG;
```
Comment (bsdjhb) [inline sys/vm/vm_fault.c:959]: Maybe move this commit before the one that removes the handling from arm64?

**Review by bsdjhb — APPROVED** [review]
Generally look ok to me, one suggestion about commit ordering

File: sys/vm/vm_fault.c (hunk 1 above)
Comment (markjdb) [reply sys/vm/vm_fault.c:959]: Yes, that makes sense. Done.

**Review by jrtc27 — APPROVED** [review]

**Comment by nwf** [comment]
FWIW, LGTM

**Review by brooksdavis — APPROVED** [review]

**Pull request merged**